cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/GcpProvisionSetup.java

```diff
@@ -146,4 +139,111 @@ private boolean containsSpecificImage(ImageList imageList, String imageUrl) {
         }
         return false;
     }
+
+    private boolean bucketExist(Storage storage, String bucketName) {
+        boolean existingBucket;
+        try {
+            storage.buckets().get(bucketName).execute();
+            existingBucket = true;
+        } catch (GoogleJsonResponseException ex) {
+            existingBucket = false;
+            if (ex.getStatusCode() == NOT_FOUND) {
+                LOGGER.warn("Bucket {} does not exist on provider side so we will create it: {}",
+                        bucketName, ex.getMessage());
+            } else {
+                LOGGER.warn("We were not able to get the bucket from Google side with name {} with exception {}. "
+                                + "We do not stop the provisioning process because the customer probably dont give us storage.get permission.",
+                        bucketName, ex.getMessage());
+            }
+        } catch (Exception e) {
+            LOGGER.warn("Unexpected error occurred when we tried to get bucket {} from Google side: {}",
+                    bucketName, e.getMessage());
+            existingBucket = true;
+        }
+        return existingBucket;
+    }
+
+    private Bucket createBucketIfNotExists(String projectId, AuthenticatedContext authenticatedContext, CloudContext cloudContext, Storage storage) {
+        String gcpLocation = authenticatedContext.getCloudContext().getLocation().getRegion().getRegionName();
+        String bucketName = String.format("%s-%s-%d", projectId, cloudContext.getName(), cloudContext.getId());
```

Comment:
the problem will be the same here cloudContext.getId() is dynamic

Reply:
yes, but we discussed this offline and agreed on to keep the original behaviour in his line of CB. Should I modify it?